src/js/redux_state/modules/registration.js

```diff
@@ -47,91 +49,113 @@ export const actions = makeActions('registration', {
-      }
-    }
-  },
-
-  generateSeedPhrase: {
-    expectedParams: ['randomString'],
-    creator: (randomString) => {
-      return (dispatch, getState) => {
-        const mnemonic = bip39.entropyToMnemonic(randomString)
-        dispatch(actions.setPassphrase({mnemonic}))
-      }
-    }
-  },
-
   generateAndEncryptKeyPairs: {
     expectedParams: [],
     async: true,
-    creator: (params) => {
+    creator: () => {
       return async (dispatch, getState, {services, backend}) => {
-        const seedPhrase = getState().getIn([
+        const randomString = getState().getIn([
           'registration',
           'passphrase',
-          'phrase'
+          'randomString'
         ])
-        if (!seedPhrase) {
-          throw new Error('No seedphrase found.')
-        }
-        const masterKeyPair = deriveMasterKeyPairFromSeedPhrase(seedPhrase)
-        console.log(masterKeyPair, 'here is our masterkeypair')
-        const genericSigningKey = deriveGenericSigningKeyPair(masterKeyPair)
 
         const password = getState().getIn([
           'registration',
           'encryption',
           'pass'
         ])
 
+        if (!randomString) {
+          // TODO consistent error handling
+          throw new Error('No random string provided')
+        }
+
+        const {
+          didDocument,
+          mnemonic,
+          masterKeyWIF,
+          genericSigningKeyWIF,
+          ethereumKeyWIF
+        } = backend.jolocomLib.identity.create(randomString)
+
+        const {privateKey, address} = cryptoUtils.decodeWIF(ethereumKeyWIF)
+
+        try {
+          await backend.ethereum.requestEther({ did: didDocument.did, address })
+        } catch (err) {
+          // TODO consistent error handling
+          throw new Error(err)
+        }
+
+        // TODO consistent error handling
+        const ddoHash = await backend.jolocomLib.identity.store(didDocument)
+
+        try {
+          await backend.jolocomLib.identity.register(
+            Buffer.from(privateKey, 'hex'),
+            didDocument.id,
+            ddoHash
+          )
+        } catch (err) {
+          throw new Error(err)
+          // TODO consistent error handling
+        }
+
         const encMaster = await backend.encryption.encryptInformation({
           password,
-          data: masterKeyPair.keyPair.toWIF()
+          data: masterKeyWIF
         })
 
         const encGeneric = await backend.encryption.encryptInformation({
           password,
-          data: genericSigningKey.keyPair.toWIF()
+          data: genericSigningKeyWIF
         })
 
         await services.storage.setItem('masterKeyWIF', encMaster)
         await services.storage.setItem('genericKeyWIF', encGeneric)
 
-        // dispatch(router.pushRoute('/wallet'))
+        dispatch(actions.setRandomString({randomString: ''}))
+        dispatch(actions.setPassphrase({mnemonic}))
+        // dispatch(actions.goForward())
+      }
+    }
+  },
+
+  publishDDO: {
```

Comment:
Is it used anywhere? `store` was called already

Reply:
You are right, removed.